frame/support/procedural/src/lib.rs

```diff
@@ -583,3 +584,646 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
+/// implemented for `Pallet` allowing access to the storage from pallet struct.
+///
+/// Thus when defining a storage named `Foo`, it can later be accessed from `Pallet` using
+/// `<Pallet as Store>::Foo`.
+#[proc_macro_attribute]
+pub fn generate_store(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// To generate the full storage info (used for PoV calculation) use the attribute
+/// `#[pallet::generate_storage_info]`, e.g.:
+///
+/// ```ignore
+/// #[pallet::pallet]
+/// #[pallet::generate_storage_info]
+/// pub struct Pallet<T>(_);
+/// ```
+///
+/// This requires all storage items to implement the trait `StorageInfoTrait`, thus all keys
+/// and value types must be bound by `MaxEncodedLen`. Individual storages can opt-out from this
+/// constraint by using [`#[pallet::unbounded]`](`macro@unbounded`) (see
+/// [`#[pallet::storage]`](`macro@storage`) for more info).
+#[proc_macro_attribute]
+pub fn generate_storage_info(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// Because the `pallet::pallet` macro implements `GetStorageVersion`, the current storage
+/// version needs to be communicated to the macro. This can be done by using the
+/// `pallet::storage_version` attribute:
+///
+/// ```ignore
+/// const STORAGE_VERSION: StorageVersion = StorageVersion::new(5);
+///
+/// #[pallet::pallet]
+/// #[pallet::storage_version(STORAGE_VERSION)]
+/// pub struct Pallet<T>(_);
+/// ```
+///
+/// If not present, the current storage version is set to the default value.
+///
+/// ### Macro expansion:
+///
+/// The macro adds this attribute to the struct definition:
+/// ```ignore
+/// #[derive(
+/// 	frame_support::CloneNoBound,
+/// 	frame_support::EqNoBound,
+/// 	frame_support::PartialEqNoBound,
+/// 	frame_support::RuntimeDebugNoBound,
+/// )]
+/// ```
+/// and replaces the type `_` with `PhantomData<T>`. It also implements on the pallet:
+/// * `GetStorageVersion`
+/// * `OnGenesis`: contains some logic to write the pallet version into storage.
+/// * `PalletErrorTypeInfo`: provides the type information for the pallet error, if defined.
+///
+/// It declares `type Module` type alias for `Pallet`, used by `construct_runtime`.
+///
+/// It implements `PalletInfoAccess` on `Pallet` to ease access to pallet information given by
+/// `frame_support::traits::PalletInfo`. (The implementation uses the associated type
+/// `frame_system::Config::PalletInfo`).
+///
+/// It implements `StorageInfoTrait` on `Pallet` which give information about all storages.
+///
+/// If the attribute `generate_store` is set then the macro creates the trait `Store` and
+/// implements it on `Pallet`.
+///
+/// If the attribute set_storage_max_encoded_len is set then the macro call `StorageInfoTrait`
+/// for each storage in the implementation of `StorageInfoTrait` for the pallet. Otherwise it
+/// implements `StorageInfoTrait` for the pallet using the `PartialStorageInfoTrait`
+/// implementation of storages.
```

Comment:
All of this above is describing the `#[pallet::pallet]` macro, not `#[pallet::storage_version]`.

Reply:
good catch! I was worried something like this slipped through.

Reply:
Do you just mean the note starting with `If the attribute set_storage_max_encoded_len` is set....`, or farther up as well? From the looks of it this all seems to be about storage stuff related to the storage version macro....?

Reply:
this is the only outstanding issue if you can give me some clarification @KiChjang

Reply:
ok fixed thanks again!